cpp/include/cudf/digitize.hpp

```diff
@@ -30,23 +30,19 @@ namespace cudf {
  * `i` such that `bins[i-1] <= x < bins[i]`. If `right == true`, it will find `i`
  * such that `bins[i - 1] < x <= bins[i]`. Finally, if `x < bins[0]` or
  * `x > bins[num_bins - 1]`, it sets the index to `0` or `num_bins`, respectively.
- *
- * NOTE: This function does not handle null values and will throw if `col`
- * or `bins` contain any.
  * 
  * @throws cudf::logic_error if `col` and `bins` types mismatch
- * @throws cudf::logic_error if type is not numeric or timestamp
- * @throws cudf::logic_error if `col` or `bins` contain any null values
  *
  * @param col column_view with the values to be binned
  * @param bins column_view of ascending bin boundaries
  * @param right Whether the intervals should include the left or right bin edge
+ * @param order Whether nulls should be sorted before or after other values
  * @param mr Optional resource to use for device memory allocation
  *
  * @returns device array of same size as `col` to be filled with bin indices
  */
 std::unique_ptr<column>
-digitize(column_view const& col, column_view const& bins, bool right,
+digitize(column_view const& col, column_view const& bins, bool right, null_order order,
```

Comment:
I think `bool right` should be changed to something like:
```
enum bound{
   upper,
   lower
};
```

Reply:
Ok, I added this to types.hpp as
```
/**---------------------------------------------------------------------------*
 * @brief Indicates whether the bound of a range is included or excluded.
 *---------------------------------------------------------------------------**/
enum class range_bound : bool {
  INCLUSIVE,  ///< Range includes values less than or equal to the bound
  EXCLUSIVE   ///< Range excludes values greater than or equal to the bound
};
```